sonaranalyzer-dotnet/src/Rspec2Resx.targets

```diff
@@ -3,23 +3,28 @@
   <PropertyGroup>
     <EnableDefaultEmbeddedResourceItems>false</EnableDefaultEmbeddedResourceItems>
   </PropertyGroup>
+
   <ItemGroup Condition="'$(Rspec2Resx)' != ''">
     <RSpecFiles Include="$(ProjectDir)..\..\rspec\$(Rspec2Resx)\*.html"/>
     <RSpecFiles Include="$(ProjectDir)..\..\rspec\$(Rspec2Resx)\*.json"/>
     <!--Must be explicitely added, because we are executed after the list has already been filled by what exists in the folder -->
     <EmbeddedResource Include="RspecStrings.resx" />
   </ItemGroup>
+
+  <!-- BeforeTargets="DispatchToInnerBuilds" is needed since we target multiple .net frameworks and the RspecStrings.resx file should be gennerated only once. //-->
   <Target Condition="'$(Rspec2Resx)' != ''"
-          Name="GenerateResxFromRspec" 
-          Inputs="@(RSpecFiles)" 
-          Outputs="RspecStrings.resx" 
-          BeforeTargets="BeforeBuild">
+          Name="GenerateResxFromRspec"
+          Inputs="@(RSpecFiles)"
+          Outputs="RspecStrings.resx"
+          BeforeTargets="DispatchToInnerBuilds;BeforeBuild">
```

Comment:
do we need both `"DispatchToInnerBuilds;BeforeBuild"` ? I see in [this thread](https://github.com/dotnet/msbuild/issues/2781) that they had the same problem and used a similar solution, but maybe in a comment it would be nice to explain why both are needed

for the future ourselves

Reply:
Actually `DispatchToInnerBuilds` is not executed at all for our build (that's why I removed you from reviewers and put the issue back in progress)

Looking at the build logs, there were 2 parallel calls:
```
2020-08-14T09:52:56.2280095Z   SonarAnalyzer.CFG -> D:\a\1\s\sonaranalyzer-dotnet\src\SonarAnalyzer.CFG\bin\Release\netstandard2.0\SonarAnalyzer.CFG.dll
2020-08-14T09:52:56.2281925Z   Sonar: (SonarAnalyzer.CFG.csproj) Project processed successfully
2020-08-14T09:52:56.2283187Z   Finding ResGen.exe path
2020-08-14T09:52:56.2284601Z   Path found: C:\Program Files (x86)\\Microsoft SDKs\\Windows\\v10.0A\\bin\\NETFX 4.8 Tools\\ResGen.exe
2020-08-14T09:52:56.2913081Z   Path found: C:\Program Files (x86)\\Microsoft SDKs\\Windows\\v10.0A\\bin\\NETFX 4.8 Tools\\ResGen.exe
2020-08-14T09:52:56.4003656Z   Retrieving rules for language: vbnet
2020-08-14T09:52:56.5906542Z   Retrieving rules for language: vbnet
2020-08-14T09:52:56.6193185Z   Creating resources
2020-08-14T09:52:57.5006478Z   Creating resources
2020-08-14T09:52:58.3395122Z   Writing raw resources at D:\a\1\s\scripts\rspec\vbnet_strings.restext
2020-08-14T09:52:58.7013225Z   Generating resx files at D:\a\1\s\sonaranalyzer-dotnet\src\SonarAnalyzer.VisualBasic\RspecStrings.resx
2020-08-14T09:52:58.9696136Z   Writing raw resources at D:\a\1\s\scripts\rspec\vbnet_strings.restext
2020-08-14T09:52:59.2298531Z   Read in 1368 resources from "D:\a\1\s\scripts\rspec\vbnet_strings.restext"
2020-08-14T09:52:59.2737479Z   Writing resource file...  Done.
2020-08-14T09:52:59.4814126Z   Generating resx files at D:\a\1\s\sonaranalyzer-dotnet\src\SonarAnalyzer.VisualBasic\RspecStrings.resx
2020-08-14T09:52:59.7273000Z   Read in 1368 resources from "D:\a\1\s\scripts\rspec\vbnet_strings.restext"
2020-08-14T09:52:59.7301556Z   Writing resource file...  Done.
```
just a matter of luck that the write didn't failed.

Reply:
I've fixed the problem by disabling parallel builds.

Reply:
Another solution would be to add the file to git, remove the target file which generates it and make sure that we update it after updating the rspecs. We already have `rspec.ps1` for that so we just need to update the script to generate also the resx.

The advantages I see with the later approach:
- it will done only during the rspec update. As far as I can tell it is the only moment when this is needed since only then the input files are updated. 
- it will be done only once (not once per target framework)
- will allow parallel builds

What do you think about this?

Reply:
It makes sense to only generate it when it's necessary - during rspec updates.

Reply:
Done in: #3508